### PR TITLE
Add google authentication

### PR DESCRIPTION
**Pull request by zenithyap**
1. Create UI for alternative authentication methods
2. Add google authentication

Authentication for others to be done on another branch.

### CODE REVIEW

File: front-end/peer-prep/src/components/Auth/SignInPage/index.tsx [hunk 1]
```diff
@@ -25,6 +30,31 @@ const SignInPage = () => {
...
+  .then((result) => {
+    // This gives you a Google Access Token. You can use it to access the Google API.
+    const credential = GoogleAuthProvider.credentialFromResult(result);
+    const token = credential?.accessToken;
+
+    // The signed-in user info.x
+    const user = result.user;
+    // IdP data available using getAdditionalUserInfo(result)
+    // ...
+
```
Comment (Acerizm) [inline front-end/peer-prep/src/components/Auth/SignInPage/index.tsx]: I would suggest if you could immediately allow the user to navigate to `/home`!

An example could be:
`const googleSignIn = () => signInWithPopup(....)
    .then((result) => {
        useNavigate("/home")
    });
`

**Review by Acerizm — CHANGES_REQUESTED** [review]
LGTM! Once you allow the user to be redirected to the homepage automatically after signing in with Google and all will be good!

File: front-end/peer-prep/src/components/Auth/SignInPage/index.tsx (hunk 1 above)
Comment (zenithyap) [reply front-end/peer-prep/src/components/Auth/SignInPage/index.tsx]: Thanks for the suggestion, I have fixed it. Let me know if it works now!

Comment (Acerizm) [reply front-end/peer-prep/src/components/Auth/SignInPage/index.tsx]: Yeap it works thanks for the change!

**Review by Acerizm — APPROVED** [review]

**Pull request merged**